### PR TITLE
Mpatelcz/influxdb publisher

### PR DESCRIPTION
**Pull request by mpatelcz**
Code rework to allow using snap publishers other than cassandra.
Swan snap sessions were reorganized/simplified.
Added support for influxdb snap publisher.

Tests done:
integration tests

### CODE REVIEW

**Comment by mpatelcz** [comment]
run integration tests

**Comment by mpatelcz** [comment]
run integration tests

**Comment by mpatelcz** [comment]
run integration tests

**Comment by mpatelcz** [comment]
run integration tests

**Comment by mpatelcz** [comment]
run integration tests

File: pkg/metadata/influxdb.go [hunk 1]
```diff
@@ -90,8 +90,10 @@ func NewInfluxDB(experimentID string, config InfluxDBConfig) (Metadata, error) {
 // influxDBStoreMap writes metadata to the database with tags attached to it.
 // It writes values (metadata) one by one/row by row. No aggregation is being done.
 func influxDBStoreMap(m *InfluxDB, metadata map[string]string, kind string) error {
-
-	//err := m.session.Query(`INSERT INTO metadata (experiment_id, kind, time, timeuuid, metadata) VALUES (?, ?, ?, ?, ?)`, m.experimentID, kind, time.Now(), gocql.TimeUUID(), metadata).Exec()
+	if len(metadata) == 0 {
```
Comment (ppalucki) [inline pkg/metadata/influxdb.go]: kill it or error

File: pkg/snap/sessions/use/use.go [hunk 2]
```diff
@@ -17,86 +17,56 @@ package use
...
-	sessions.ApplyCassandraConfiguration(publisher)
+func DefaultConfig() snap.SessionConfig {
+	pub := publishers.NewDefaultPublisher()
 
-	return Config{
+	return snap.SessionConfig{
 		SnapteldAddress: snap.SnapteldAddress.Value(),
 		Interval:        1 * time.Second,
-		Publisher:       publisher,
+		Publisher:       pub.Publisher,
```
Comment (ppalucki) [inline pkg/snap/sessions/use/use.go]: pub.Publisher can be moved tp pkg.snap

**Pull request merged**